solana-programs/staking-bridge/programs/staking-bridge/src/lib.rs

```diff
@@ -1,15 +1,271 @@
+    #[account(mut)]
+    /// CHECK: ok
+    pub user_source_token_account: AccountInfo<'info>,
+    #[account(mut)]
+    /// CHECK: ok
+    pub user_destination_token_account: AccountInfo<'info>,
+    #[account(
+        seeds = [b"staking_bridge".as_ref()],
+        bump = staking_bridge_pda_bump
+    )]
+    /// CHECK: This is the PDA initialized in the CreatePDA instruction.
+    pub user_source_owner: AccountInfo<'info>,
+    pub spl_token_program: Program<'info, Token>,
+}
+
+#[derive(AnchorDeserialize, AnchorSerialize, Default)]
+pub struct PostWormholeMessageData {
+    pub nonce: u32,
+    pub amount: u64,
+    pub fee: u64,
+    pub target_address: [u8; 32],
+    pub target_chain: u16,
+}
+
+#[derive(Accounts)]
+#[instruction(
+    _nonce: u32,
+    _amount: u64,
+    _fee: u64,
+    _config_bump: u8,
+    _wrapped_mint_bump: u8,
+    _wrapped_meta_bump: u8,
+    _authority_signer_bump: u8,
+    _bridge_config_bump: u8,
+    _emitter_bump: u8,
+    _sequence_bump: u8,
+    _fee_collector_bump: u8,
+    staking_bridge_pda_bump: u8
+)]
+pub struct PostWormholeMessage<'info> {
+    /// CHECK: This is the Token Bridge program id
+    pub program_id: AccountInfo<'info>,
+    /// CHECK: This is the Core Bridge program id
+    pub bridge_id: AccountInfo<'info>,
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    #[account()]
+    /// CHECK: This is the config PDA owned by the Token Bridge
+    pub config: AccountInfo<'info>,
+    #[account(mut)]
+    /// CHECK: This is the wrapped mint PDA, which also depends on the origin token chain and origin token address, owned by the Token Bridge
+    pub wrapped_mint: AccountInfo<'info>,
+    #[account()]
+    /// CHECK: This is the wrapped meta PDA, which also depends on the wrapped mint, owned by the Token Bridge
+    pub wrapped_meta: AccountInfo<'info>,
+    #[account()]
+    /// CHECK: This is the authority signer PDA owned by the Token Bridge
+    pub authority_signer: AccountInfo<'info>,
+    #[account(mut)]
+    /// CHECK: This is the bridge PDA owned by the Core Bridge
+    pub bridge_config: AccountInfo<'info>,
+    #[account()]
+    /// CHECK: This is the emitter PDA owned by the Token Bridge
+    pub emitter: AccountInfo<'info>,
+    #[account(mut)]
+    /// CHECK: This is the sequence PDA, which also depends on the emitter, owned by the Core Bridge
+    pub sequence: AccountInfo<'info>,
+    #[account(mut)]
+    /// CHECK: This is the fee collector PDA owned by the Core Bridge
+    pub fee_collector: AccountInfo<'info>,
+    #[account(mut)]
+    pub message: Signer<'info>,
+    #[account(
+        mut,
+        seeds = [b"staking_bridge".as_ref()],
+        bump = staking_bridge_pda_bump
+    )]
+    /// CHECK: This is the PDA initialized in the CreatePDA instruction.
+    pub from_owner: AccountInfo<'info>,
+    #[account(mut)]
+    /// CHECK: This is the associated token account of the PDA, from which the tokens will be transferred
+    pub from: AccountInfo<'info>,
```

Comment:
I think for some of these we might want to use `TokenAccount` etc? https://docs.rs/anchor-spl/latest/anchor_spl/token/struct.TokenAccount.html

Reply:
nice. will try that

Reply:
Doesnt work... Using UncheckedAccount